src/coreclr/tools/Common/Internal/Text/Utf8String.cs

```diff
@@ -72,67 +71,28 @@ public override unsafe int GetHashCode()
 
         public bool Equals(Utf8String other)
         {
-            int length = _value.Length;
-            if (length != other.Length)
-                return false;
-
-            if (_value == other._value)
-                return true;
-
-            unsafe
-            {
-                fixed (byte* ap = _value) fixed (byte* bp = other._value)
-                {
-                    byte* a = ap;
-                    byte* b = bp;
-
-                    while (length >= 4)
-                    {
-                        if (*(int*)a != *(int*)b) return false;
-                        a += 4; b += 4; length -= 4;
-                    }
-                    if (length >= 2)
-                    {
-                        if (*(short*)a != *(short*)b) return false;
-                        a += 2; b += 2; length -= 2;
-                    }
-                    if (length > 0)
-                    {
-                        if (*a != *b) return false;
-                    }
-                    return true;
-                }
-            }
+            return UnderlyingArray.SequenceEqual(other.UnderlyingArray);
         }
 
         private static int Compare(Utf8String strA, Utf8String strB)
         {
-            int length = Math.Min(strA.Length, strB.Length);
+            ReadOnlySpan<byte> x = strA.UnderlyingArray;
+            ReadOnlySpan<byte> y = strB.UnderlyingArray;
+
+            int length = Math.Min(x.Length, y.Length);
+
+            int diffIndex = x.CommonPrefixLength(y);
 
-            unsafe
+            if (diffIndex != length)
             {
-                fixed (byte* ap = strA._value)
-                fixed (byte* bp = strB._value)
-                {
-                    byte* a = ap;
-                    byte* b = bp;
-
-                    while (length > 0)
-                    {
-                        if (*a != *b)
-                            return *a - *b;
-                        a += 1;
-                        b += 1;
-                        length -= 1;
-                    }
-
-                    // At this point, we have compared all the characters in at least one string.
-                    // The longer string will be larger.
-                    // We could optimize and compare lengths before iterating strings, but we want
-                    // Foo and Foo1 to be sorted adjacent to eachother.
-                    return strA.Length - strB.Length;
-                }
+                return x[diffIndex] - y[diffIndex];
```

Comment:
You could also just forward this to `span.SequenceCompareTo(other)`.

Reply:
Hmm, yeah that _seems_ to be better. Thanks!

I also noticed that CommonPrefixLength is not being benched in dotnet/performance 🤔